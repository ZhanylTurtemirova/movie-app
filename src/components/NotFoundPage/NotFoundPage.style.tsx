import styled from "styled-components";

export const ContentWrapper = styled.div`
  width: 50%;
  height: 50%;
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const Wrapper = styled.div`
  width: 100%;
  height: 100vh;
  margin: 10px auto;
  background: var(--mainDark);
  display: flex;
  flex-direction: column;
`;

export const PageTitle = styled.h1`
  text-align: center;
  font-size: 20px;
`;
export const PageImg = styled.img`
  width: 100%;
  height: auto;
`;
