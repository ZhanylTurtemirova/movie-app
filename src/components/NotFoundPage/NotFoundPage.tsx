import React, { FC } from "react";

import { Wrapper, PageImg, PageTitle } from "./NotFoundPage.style";
import Logo from "../Logo";
import notFound from "../../assets/notFound.png";
import Footer from "../Footer";
import Button from "../Button";

const NotFoundPage: FC = (): React.ReactElement => (
  <Wrapper>
    <Logo />
    <PageTitle>Page Not Found</PageTitle>
    <PageImg src={notFound} />
    <Button text="Go back" />
    <Footer />
  </Wrapper>
);
export default NotFoundPage;
