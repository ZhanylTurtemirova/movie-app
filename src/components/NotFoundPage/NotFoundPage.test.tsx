import { fireEvent, render, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import NotFoundPage from "./NotFoundPage";
import { MemoryRouter } from "react-router-dom";
const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));
describe("NotFoundPage", () => {
  it("should render", () => {
    const { asFragment } = render(<NotFoundPage />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should find title", () => {
    const { getByText } = render(<NotFoundPage />);
    expect(getByText("Page Not Found")).toBeInTheDocument();
  });
  it("should find and click Back button", async () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );
    const backBtn = getByTestId("backBtn");
    expect(backBtn).toBeInTheDocument();
    act(() => {
      fireEvent.click(backBtn);
    });
    await waitFor(() => {
      expect(mockHistoryPush).toHaveBeenCalledWith("/");
    });
  });
});
