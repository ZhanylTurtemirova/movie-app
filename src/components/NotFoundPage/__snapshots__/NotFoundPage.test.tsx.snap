// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render 1`] = `
<DocumentFragment>
  <div
    class="sc-gsDJrp elmpvz"
  >
    <div>
      <span
        class="sc-gKckTs daCzoc"
      >
        netflix
      </span>
      <span
        class="sc-jRQAMF fYDzkP"
      >
        roulette
      </span>
    </div>
    <div
      class="sc-bdvvaa jKKvaL"
    >
      <h1
        class="sc-dkPtyc fUjyaa"
      >
        Page Not Found
      </h1>
      <img
        class="sc-hKwCoD cQBGXu"
        src="notFound.png"
      />
      <button
        class="sc-eCImvq bytteG"
        data-testid="backBtn"
      >
        Go back
      </button>
    </div>
    <div
      class="sc-iCfLBT dDymkj"
    >
      <div>
        <span
          class="sc-gKckTs daCzoc"
        >
          netflix
        </span>
        <span
          class="sc-jRQAMF fYDzkP"
        >
          roulette
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
